tasks/rh-sign-cosign-image/rh-sign-cosign-image.yaml

```diff
@@ -0,0 +1,103 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: rh-sign-cosign-image
+  labels:
+    app.kubernetes.io/version: "0.1.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Task to sign container images in a Snapshot spec using the cosign.
+  params:
+    - name: snapshotPath
+      description: Path to the JSON string of the mapped Snapshot spec in the data workspace
+      type: string
+    - name: dataPath
+      description: Path to the JSON string of the merged data to use in the data workspace
+      type: string
+    - name: secretName
+      description: Name of secret containing needed credentials 
+      type: string
+  workspaces:
+    - name: data
+      description: workspace to read and save files
+  steps:
+    - name: sign-image
+      image: quay.io/konflux-ci/release-service-utils:38c3bfd7479c86b832cba5b61f9bbde40c469393
+      env:
+        - name: AWS_REGION
+          valueFrom:
+            secretKeyRef:
+              name: $(params.secretName)
+              key: AWS_REGION
+        - name: AWS_ACCESS_KEY_ID
+          valueFrom:
+            secretKeyRef:
+              name: $(params.secretName)
+              key: AWS_ACCESS_KEY_ID
+        - name: AWS_SECRET_ACCESS_KEY
+          valueFrom:
+            secretKeyRef:
+              name: $(params.secretName)
+              key: AWS_SECRET_ACCESS_KEY
+      script: |
+        #!/usr/bin/env bash
+        set -eu
+        function fetch_cosign_key() {
+          echo $(kubectl_get_arg_from_config_map $1 SIG_KEY)
+        } 
+
+        SNAPSHOT_PATH=$(workspaces.data.path)/$(params.snapshotPath)
+
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No valid data file was provided."
+            exit 1
+        fi
+
+        request=$(jq -r '.sign.request // "hacbs-signing-pipeline"' ${DATA_FILE})
+        CONFIG_MAP_NAME=$(jq -rc '.sign.configMapName // "signing-config-map"' ${DATA_FILE})
+
+        COMPONENTS_LENGTH=$(jq '.components |length' ${SNAPSHOT_PATH})
+
+        COSIGN_SIG_KEY=$(fetch_cosign_key $CONFIG_MAP_NAME)
```

Comment:
Again, I'm not sure what this is supposed to do (it currently does nothing), but since it's a secret, it should probably use a secret and not a configmap. But looking at cosign docs: https://github.com/sigstore/cosign/blob/main/doc/cosign_sign.md

> path to the private key file, KMS URI or Kubernetes Secret

So maybe you can pass a secret name directly and it will fetch it? Not sure what the expected key in the secret would be though. Otherwise a file can be prepared.

Reply:
I see. So in this case it's some kind of aws path that is used to fetch the actual key. Then it sounds good.